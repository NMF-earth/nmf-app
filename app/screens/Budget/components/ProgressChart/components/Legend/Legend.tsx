import React from "react";
import { View } from "react-native";

import styles from "./Legend.styles";
import LegendItem from "../LegendItem";
import { Colors } from "../../../../../../style";
import { t } from "../../../../../../utils";

interface Prop {
  totalEmissions: number;
  foodEmissions: number;
  transportEmissions: number;
  otherEmissions: number;
}

const Legend = ({
  totalEmissions,
  transportEmissions,
  foodEmissions,
  otherEmissions,
}: Prop) => {
  const items = [
    {
      name: t("BUDGET_SCREEN_PROGRESS_CHART_LEGEND_ITEM_TOTAL"),
      value: totalEmissions,
      color: Colors.apricot,
    },
    {
      name: t("BUDGET_SCREEN_PROGRESS_CHART_LEGEND_ITEM_TRANSPORT"),
      value: transportEmissions,
      color: Colors.yellow50,
    },
    {
      name: t("BUDGET_SCREEN_PROGRESS_CHART_LEGEND_ITEM_FOOD"),
      value: foodEmissions,
      color: Colors.green50,
    },
    {
      name: t("BUDGET_SCREEN_PROGRESS_CHART_LEGEND_ITEM_OTHER"),
      value: otherEmissions,
      color: Colors.grey70,
    },
  ];
  return (
    <View style={styles.container}>
      {items.map((item, index) => (
        <LegendItem
          key={index}
          name={item.name}
          value={item.value}
          color={item.color}
        />
      ))}
    </View>
  );
};

export default Legend;
