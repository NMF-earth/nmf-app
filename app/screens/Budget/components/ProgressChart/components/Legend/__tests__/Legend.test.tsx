import React from "react";
import { create } from "react-test-renderer";

import Legend from "../Legend";

const props = {
  totalEmissions: 900,
  foodEmissions: 200,
  transportEmissions: 600,
  otherEmissions: 100,
};

jest.unmock("../Legend");

it("Legend renders correctly", () => {
  const tree = create(<Legend {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
