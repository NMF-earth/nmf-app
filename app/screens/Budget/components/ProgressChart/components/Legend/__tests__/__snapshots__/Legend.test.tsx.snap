// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend renders correctly 1`] = `
<View
  style={
    {
      "justifyContent": "space-between",
      "paddingVertical": 20,
    }
  }
>
  <LegendItem
    amount={200}
    name="Food"
    totalAmount={1000}
  />
  <LegendItem
    amount={300}
    name="Meal"
    totalAmount={1000}
  />
  <LegendItem
    amount={200}
    name="Transport"
    totalAmount={1000}
  />
  <LegendItem
    amount={50}
    name="Streaming"
    totalAmount={1000}
  />
  <LegendItem
    amount={50}
    name="Purchase"
    totalAmount={1000}
  />
  <LegendItem
    amount={100}
    name="Fashion"
    totalAmount={1000}
  />
  <LegendItem
    amount={50}
    name="Electricity"
    totalAmount={1000}
  />
  <LegendItem
    amount={30}
    name="Product Scanned"
    totalAmount={1000}
  />
  <LegendItem
    amount={20}
    name="Custom"
    totalAmount={1000}
  />
  <LegendItem
    amount={1000}
    name="Total"
    totalAmount={1000}
  />
</View>
`;
